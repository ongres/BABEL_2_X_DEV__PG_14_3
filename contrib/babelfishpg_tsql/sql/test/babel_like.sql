set babelfishpg_tsql.sql_dialect = 'tsql';
select relname from pg_class where relname like '[';
select relname from pg_class where relname like ']';
select relname from pg_class where relname like '[]';
select relname from pg_class where relname like NULL;
select relname from pg_class where relname like '';
select relname from pg_class where relname like 'pg[1:9]class';
select relname from pg_class where relname like 'pg\[1:9\]class';
select relname from pg_class where relname like 'pg\[1:9 ]class';
select relname from pg_class where relname like 'pg [1:9\]class';

select relname from pg_class where relname like 'pg*[1:9*]class' escape '*';
select relname from pg_class where relname like 'pg [1:9*]class' escape '*';
select relname from pg_class where relname like 'pg*[1:9 ]class' escape '*';

set babelfishpg_tsql.sql_dialect = 'postgres';
select relname from pg_class where relname like '[';
select relname from pg_class where relname like ']';
select relname from pg_class where relname like '[]';
select relname from pg_class where relname like NULL;
select relname from pg_class where relname like '';
select relname from pg_class where relname like 'pg[1:9]class';
select relname from pg_class where relname like 'pg\[1:9\]class';
select relname from pg_class where relname like 'pg\[1:9 ]class';
select relname from pg_class where relname like 'pg [1:9\]class';

select relname from pg_class where relname like 'pg*[1:9*]class' escape '*';
select relname from pg_class where relname like 'pg [1:9*]class' escape '*';
select relname from pg_class where relname like 'pg*[1:9 ]class' escape '*';
