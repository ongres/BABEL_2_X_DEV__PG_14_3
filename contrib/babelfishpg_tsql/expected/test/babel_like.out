set babelfishpg_tsql.sql_dialect = 'tsql';
select relname from pg_class where relname like '[';
ERROR:  pattern matching operators '[' and ']' are not supported for LIKE
select relname from pg_class where relname like ']';
ERROR:  pattern matching operators '[' and ']' are not supported for LIKE
select relname from pg_class where relname like '[]';
ERROR:  pattern matching operators '[' and ']' are not supported for LIKE
select relname from pg_class where relname like NULL;
 relname 
---------
(0 rows)

select relname from pg_class where relname like '';
 relname 
---------
(0 rows)

select relname from pg_class where relname like 'pg[1:9]class';
ERROR:  pattern matching operators '[' and ']' are not supported for LIKE
select relname from pg_class where relname like 'pg\[1:9\]class';
 relname 
---------
(0 rows)

select relname from pg_class where relname like 'pg\[1:9 ]class';
ERROR:  pattern matching operators '[' and ']' are not supported for LIKE
select relname from pg_class where relname like 'pg [1:9\]class';
ERROR:  pattern matching operators '[' and ']' are not supported for LIKE
select relname from pg_class where relname like 'pg*[1:9*]class' escape '*';
 relname 
---------
(0 rows)

select relname from pg_class where relname like 'pg [1:9*]class' escape '*';
ERROR:  pattern matching operators '[' and ']' are not supported for LIKE
select relname from pg_class where relname like 'pg*[1:9 ]class' escape '*';
ERROR:  pattern matching operators '[' and ']' are not supported for LIKE
set babelfishpg_tsql.sql_dialect = 'postgres';
select relname from pg_class where relname like '[';
 relname 
---------
(0 rows)

select relname from pg_class where relname like ']';
 relname 
---------
(0 rows)

select relname from pg_class where relname like '[]';
 relname 
---------
(0 rows)

select relname from pg_class where relname like NULL;
 relname 
---------
(0 rows)

select relname from pg_class where relname like '';
 relname 
---------
(0 rows)

select relname from pg_class where relname like 'pg[1:9]class';
 relname 
---------
(0 rows)

select relname from pg_class where relname like 'pg\[1:9\]class';
 relname 
---------
(0 rows)

select relname from pg_class where relname like 'pg\[1:9 ]class';
 relname 
---------
(0 rows)

select relname from pg_class where relname like 'pg [1:9\]class';
 relname 
---------
(0 rows)

select relname from pg_class where relname like 'pg*[1:9*]class' escape '*';
 relname 
---------
(0 rows)

select relname from pg_class where relname like 'pg [1:9*]class' escape '*';
 relname 
---------
(0 rows)

select relname from pg_class where relname like 'pg*[1:9 ]class' escape '*';
 relname 
---------
(0 rows)

